packages/client/src/helpers/client/interceptors/authentication/token-providers/TokenData.ts

```diff
@@ -8,24 +10,24 @@ class TokenData {
    */
   static REFRESH_TOKEN_WINDOW_OFFSET = 30000;
 
+  accessToken?: string;
+  expiresIn?: string;
+  expiresTimeUtc?: number;
+  refreshToken?: string;
+  userId?: number;
+
   /**
    * Constructs a new TokenData instance with the passed in data.
    * If expiresTimeUtc is not provided, it is assumed that the token was just created and will expire after now + expiresIn seconds.
    *
    * @param {object} data - Token data.
    * @param {string} data.accessToken - The access token.
-   * @param {number} data.expiresIn - The number of seconds until the token is valid.
+   * @param {string} data.expiresIn - The number of seconds until the token is valid.
    * @param {string} [data.refreshToken] - The refresh token. Can be undefined for guest users token data.
    * @param {number} [data.expiresTimeUtc] - The calculated date in utc when the token will be expired. If not provided, it will be assumed that the token was just created and the expiresTimeUtc will be calculated from the expiresIn value based on this assumption.
    * @param {number} [data.userId] - The user id associated with this token data. Can be undefined as it is not available at the same time the token data is.
    */
-  constructor(data) {
-    if (!data) {
```

Comment:
This check should not be removed as it can still happen at runtime if the consumer is not using typescript (or simply uses a cast to force an invalid value to the constructor)

Reply:
The consumer needs to use typescript in the next version, since the blackout next version won't work properly without the needed configurations to support it, since it's a mandatory prop it should throw even when the typescript is finally converted into JS.